### PR TITLE
Refs [TASK][34567] Get data and bind data to recyclerview

### PR DESCRIPTION
**Pull request by phong10101998**
## Related Tickets
- [#Task 34567: Get data and bind data to recyclerview](https://edu-redmine.sun-asterisk.vn/issues/34567)
## WHAT
- Get data and bind data to recycler view
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/spoonacular/ui/base/MainMenuViewPagerAdapter.kt [hunk 1]
```diff
@@ -10,15 +10,16 @@ import com.sun.spoonacular.utils.MenuNumber
...
+    private val recipesRecyclerView: ArrayList<Recipe>
 ) :
     FragmentPagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
 
     override fun getCount() = MENU_COUNT
 
     override fun getItem(position: Int) = when (position) {
         MenuNumber.SEARCH.pageNumber -> SearchFragment.newInstance()
-        MenuNumber.HOME.pageNumber -> HomeFragment.newInstance(recipes)
+        MenuNumber.HOME.pageNumber -> HomeFragment.newInstance(recipesSlide,recipesRecyclerView)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/base/MainMenuViewPagerAdapter.kt]: reformat code

File: app/src/main/java/com/sun/spoonacular/ui/home/HomeFragment.kt [hunk 2]
```diff
@@ -22,9 +23,19 @@ import kotlinx.android.synthetic.main.fragment_home.*
 class HomeFragment : Fragment() {
 
     private var recipeSlide = arrayListOf<Recipe>()
+    private var recipeRecyclerView: ArrayList<Recipe> = arrayListOf()
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/home/HomeFragment.kt]: refactor code

File: app/src/main/java/com/sun/spoonacular/ui/splash/SplashViewModel.kt [hunk 3]
```diff
@@ -2,18 +2,22 @@ package com.sun.spoonacular.ui.splash
...
 
     private val repository = Repository.getInstance()
-
+    private var recipes = arrayListOf<Response<RecipeResponse>>()
     fun getRecipe() = liveData(Dispatchers.IO) {
         emit(Resource.loading(data = null))
         try {
-            emit(Resource.success(data = repository.getRecipe()))
+            recipes.add(repository.getRecipe())
+            recipes.add(repository.getRecipes())
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/splash/SplashViewModel.kt]: sao lại add 2 lần thế này e?

File: app/src/main/java/com/sun/spoonacular/ui/home/recyclerview/RecipeViewHolder.kt [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+class RecipeViewHolder(
+    itemView: View,
+    private val onItemClicked: (Recipe) -> Unit
+) : RecyclerView.ViewHolder(itemView) {
+
+    fun binData(recipe: Recipe) {
+        itemView.apply {
+            Glide.with(context)
+                .load(recipe.image)
+                .into(imgDish)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/home/recyclerview/RecipeViewHolder.kt]: viết ext load image đi

File: app/src/main/java/com/sun/spoonacular/ui/home/recyclerview/RecipeAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package com.sun.spoonacular.ui.home.recyclerview
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.sun.spoonacular.R
+import com.sun.spoonacular.data.model.Recipe
+
+class RecipeAdapter(private val onItemClicked: (Recipe) -> Unit) :
+    RecyclerView.Adapter<RecipeViewHolder>() {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/spoonacular/ui/home/recyclerview/RecipeAdapter.kt]: chuyển qua dùng ListAdapter + DiffUtil đi nhé

File: app/src/main/java/com/sun/spoonacular/ui/splash/SplashViewModel.kt (hunk 3 above)
Comment (phong10101998) [reply app/src/main/java/com/sun/spoonacular/ui/splash/SplashViewModel.kt]: > sao lại add 2 lần thế này e?

2 cái response khác nhau á a

File: app/src/main/java/com/sun/spoonacular/utils/FragmentExt.kt [hunk 6]
```diff
@@ -32,3 +35,9 @@ fun Context.hideKeyboard(view: View) {
     val inputMethodManager = getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
     inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
 }
+
+fun loadImage(context: Context, imageApi: String, imageView: ImageView) {
+    Glide.with(context)
+        .load(imageApi)
+        .into(imageView)
+}
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/spoonacular/utils/FragmentExt.kt]: sao lại viết vào Fragment Extension 
Tạo 1 file ImageExt 
và viết function Extension cho ImageView 
Image.loadUrl(...)

File: app/src/main/res/layout/fragment_home.xml [hunk 7]
```diff
@@ -15,4 +15,32 @@
...
+        android:textColor="@color/color_F28F73"
+        android:textSize="@dimen/sp_12"
+        android:textStyle="normal"
+        app:layout_constraintBottom_toTopOf="@id/recyclerViewHome"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/viewPagerSlideHome" />
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerViewHome"
+        android:layout_width="match_parent"
```
Comment (huytq-2422) [inline app/src/main/res/layout/fragment_home.xml]: match_constraint

File: app/src/main/java/com/sun/spoonacular/utils/ImageExt.kt [hunk 8]
```diff
@@ -0,0 +1,11 @@
+package com.sun.spoonacular.utils
+
+import android.content.Context
+import android.widget.ImageView
+import com.bumptech.glide.Glide
+
+fun ImageView.loadUrl(context: Context, imageApi: String) {
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/spoonacular/utils/ImageExt.kt]: mỗi view đều chứa context cho riêng nó nên mình sẽ lấy context ở ImageView luôn 
```suggestion
fun ImageView.loadUrl(url: String) {
```

File: app/src/main/java/com/sun/spoonacular/utils/ImageExt.kt [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+    Glide.with(context)
+        .load(imageApi)
+        .into(this)
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/spoonacular/utils/ImageExt.kt]: nên thêm các trường hợp load lỗi, placeholder để tăng UX 
```suggestion
        .load(imageApi)
        .error()
        .placeHolder()
        .into(this)
```

File: app/src/main/res/layout/item_recipe_home.xml [hunk 10]
```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="@dimen/dp_350"
```
Comment (huytq-2422) [inline app/src/main/res/layout/item_recipe_home.xml]: nên fix theo kích thước màn hình 
https://stackoverflow.com/a/51224889

**Review by huytq-2422 — APPROVED** [review]

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**